serde-support/src/main/java/io/micronaut/serde/support/serializers/SerBean.java

```diff
@@ -272,26 +271,50 @@ public int getOrder() {
                         final String propertyName = NameUtils.getPropertyNameForGetter(jsonGetter.getName());
                         String n = resolveName(annotationMetadata, jsonGetterAnnotationMetadata, propertyName, false, propertyNamingStrategy);
                         final Argument<Object> returnType = jsonGetter.getReturnType().asArgument();
-                        writeProperties.add(new SerProperty<>(this, n,
-                                                              returnType,
-                                                              null,
-                                                              jsonGetter,
-                                                              null,
-                                                              findSerializer(encoderContext, returnType,
-                                                                             jsonGetterAnnotationMetadata),
-                                                              null,
-                                                              encoderContext
-                                            )
+                        writePropertiesInitializer.add(
+                                () -> new SerProperty<>(SerBean.this, n,
+                                        returnType,
+                                        null,
+                                        jsonGetter,
+                                        null,
+                                        findSerializer(encoderContext, returnType, jsonGetterAnnotationMetadata),
+                                        null,
+                                        encoderContext
+                                )
                         );
                     }
-                } else {
-                    writeProperties = Collections.emptyList();
                 }
-                this.anyGetter = anyGetterProperty;
+                this.anyGetterInitializer = anyGetterPropertyInitializer;
                 this.wrapperProperty = introspection.stringValue(SerdeConfig.class, SerdeConfig.WRAPPER_PROPERTY).orElse(null);
             }
         }
-        initialized = true;
+    }
+
+    public synchronized void initialize() throws SerdeException {
```

Comment:
Can this synchronization be moved to as synchronized block instead of synchronizing on the whole method?

Reply:
Made double check locking